core/src/test/java/org/apache/iceberg/avro/TestAvroSchemaProjection.java

```diff
@@ -150,4 +150,58 @@ public void projectWithMapSchemaChanged() {
         .as("Result of buildAvroProjection is missing some IDs")
         .isFalse();
   }
+
+  @Test
+  public void projectWithVariantSchemaChanged() {
+    final org.apache.avro.Schema currentAvroSchema =
```

Comment:
Can you please remove the `final` keywords? They are unnecessary and discouraged in the Iceberg codebase.